analysis/estimate/notebooks/compare-estimate-consensus.Rmd

```diff
@@ -0,0 +1,340 @@
+---
+title: "Compare ESTIMATE results to consensus cell types"
+author: "Stephanie J. Spielman"
+date: "`r Sys.Date()`"
+output:
+  html_notebook:
+    toc: true
+    toc_depth: 3
+    code_folding: hide
+---
+
+The purpose of this notebook is to compare ESTIMATE scores to consensus cell types.
+
+## Setup
+
+
+```{r setup}
+renv::load()
+
+library(ggplot2)
+theme_set(theme_bw())
+options(readr.show_col_types = FALSE)
+```
+
+### Paths
+
+```{r paths}
+data_dir <- here::here("analysis", "estimate", "data")
+celltype_dir <- file.path(data_dir, "consensus-celltypes")
+
+library_metadata_file <- here::here("s3_files", "scpca-library-metadata.tsv")
+estimate_file <- here::here("analysis", "estimate", "results", "estimate_results.tsv")
+immune_celltypes_groups_url <- "https://raw.githubusercontent.com/AlexsLemonade/OpenScPCA-analysis/refs/heads/main/analyses/cell-type-consensus/references/consensus-immune-cell-types.tsv"
+
+
+# we'll name these by library id to start, and then swap to sample later
+celltype_files <- list.files(
+  celltype_dir, 
+  pattern = "_processed_consensus-cell-types\\.tsv\\.gz$", 
+  full.names = TRUE, 
+  recursive = TRUE
+) |>
+    purrr::set_names(
+    \(x) {stringr::str_split_i(basename(x), pattern = "_", i = 1)}
+  )
+```
+
+### Read input data
+
+```{r warning = FALSE}
+estimate_df <- readr::read_tsv(estimate_file)
+library_metadata_df <- readr::read_tsv(library_metadata_file)
+
+id_df <- estimate_df |>
+  dplyr::inner_join(
+    library_metadata_df |>
+      dplyr::filter(seq_unit %in% c("cell", "nucleus")) |>
+      dplyr::select(scpca_sample_id, scpca_library_id),
+    by = c("sample_id" = "scpca_sample_id")
+  ) |>
+  dplyr::select(project_id, sample_id, library_id = scpca_library_id) |>
+  dplyr::distinct()
+
+library_sample_c <- id_df$sample_id |>
+  purrr::set_names(id_df$library_id)
+
+celltype_data <- celltype_files |>
+  # only keep libraries of interest
+  purrr::keep_at(
+    \(library_id) library_id %in% names(library_sample_c)
+  ) |> 
+  purrr::map(
+    \(f) {
+      readr::read_tsv(f) |>
+        dplyr::select(project_id, sample_id, consensus_annotation, consensus_ontology)
+  })
+
+# update names to sample ids
+sample_names <- library_sample_c[names(celltype_data)] |> unname()
+names(celltype_data) <- sample_names
+
+immune_df <- readr::read_tsv(immune_celltypes_groups_url)
+```
+
+### Define cell type groups
+
+The first step is to group consensus cell types into "immune" and "stromal" categories to enable comparisons with ESTIMATE scores.
+
+Let's first look at what the consensus cell types that we have for this data are.
+For human readability, we'll look at the annotations, not ontologies, for this.
+
+```{r}
+all_celltypes <- celltype_data |>
```

Comment:
The rendered copy has the list printing out twice, but I'm not sure why given the code. Also can you sort these alphabetically?

Reply:
> The rendered copy has the list printing out twice, but I'm not sure why given the code.

This is a long-standing very irritating bug with rendering rmarkdown notebooks. Like, it's all over our rendered training notebooks too 😭 😭 😭 😭

Reply:
But yes, will sort!!

Reply:
> > The rendered copy has the list printing out twice, but I'm not sure why given the code.
> 
> This is a long-standing very irritating bug with rendering rmarkdown notebooks. Like, it's all over our rendered training notebooks too 😭 😭 😭 😭

Ah that makes sense! I do remember that being an issue.